R/annotation-data-helper.R

```diff
@@ -78,13 +82,20 @@ getReducedExonRanges <- function(exonRanges.firstExon.byGene, numberOfCores = 1)
   print('Identify overlapping first exons for each gene and annotate with the promoter ids...')
   if (numberOfCores == 1) {
     exonReducedRanges.list <- lapply(1:length(exonRanges.firstExon.byGene), reduceExonsByGene, exonRanges.firstExon.byGene)
-  } else {
+  } else if (numberOfCores > 1 & .Platform$OS.type != 'windows') {
```

Comment:
instead of calculating this differently depending on the system, you could try the following which works on unix (and might work on windows as well):
  bpParameters <- BiocParallel::bpparam()
  bpParameters$workers <- numberOfCores
 exonReducedRanges.list <- BiocParallel::bplapply(1:length(exonRanges.firstExon.byGene), reduceExonsByGene, exonRanges.firstExon.byGene, BPPARAM = bpParameters)